Buildkite/ephemeral-agent.md

```diff
@@ -0,0 +1,119 @@
+
+You will later use this base image to create a VM config (a container template) for the ephemeral agent.
+
+1. Set up a new Orka VM. You can set up an Orka VM using the Orka [CLI][cli] or [REST API][api]. For more information, see the Orka [quick start guide][quick-start].  
+2. Connect to the Orka VM using VNC.  
+**Note**: The VM IP and VNC ports are displayed once the VM is deployed in Orka.  
+3. Verify that [Homebrew][homebrew] is installed on the VM. To install Homebrew, see [here][homebrew].
+4. Install the Buildkite agent on the VM by running:  
+    ```
+    brew tap buildkite/buildkite
+    brew install buildkite-agent
+    ```
+5. Verify that SSH login with a private key is enabled. SSH login is used by the wrapper agent to communicate with the ephemeral agent.
+6. On your local machine, run `orka image save`. The command saves the base image in Orka.
+
+## Set up an Orka VM config for the ephemeral agents
+
+To allow the wrapper agent to spin up ephemeral agents in Orka, create an Orka VM config (a container template) that uses the Buildkite-enabled base image you just created.  
+
+You can create an Orka VM config using the Orka [CLI][cli] or [REST API][api]. For more information, see the Orka [quick start guide][quick-start].
+
+## Set up the Buildkite wrapper agent
+
+1. Install and configure a Buildkite agent on a machine that has network visibility to your Orka environment. For more information about how to install and configure a Buildkite agent, see the [instructions][agent-instructions] provided by Buildkite.
+2. Copy the provided scripts to the agent machine: [base.sh](scripts/base.sh), [bootstrap.sh](scripts/bootstrap.sh), [run.sh](scripts/run.sh). 
+**Note**: All scripts should be in the same directory. For example, you can add them to `/usr/local/var/buildkite-agent/`. Overwrite the existing `bootstrap.sh` file if asked.
+3. Make the scripts executable by running `chmod +x path_to_script` in the command line.
+4. Copy the provided [pre-exit hook](scripts/pre-exit) to the agent machine. You should add it to the agent hooks directory. It can be found in the `buildkite-agent.cfg` file under the `hooks-path` property. The location of the `buildkite-agent.cfg` varies depending on your OS. For more information, see your [platform's installation instructions][agent-instructions].
+5. Make the hook executable by running `chmod +x path_to_hook` in the command line.
+6. Verify [jq][jq] is installed. For more information, see [jq][jq] page.
+7. Verify that the machine has network visibility to the Orka environment. If the machine is part of the Orka environment, skip this step. You can use any VPN client to connect to the Orka environment. For more information, see your Orka [IP Plan][ip-plan].
+8. Verify that the private SSH key created [earlier](#set-up-an-orka-vm-base-image) that is used to connect to the ephemeral agent is present on the wrapper machine and added to the ssh-agent.
```

Comment:
Verify that the private SSH key for connecting to the ephemeral agent is present on the wrapper machine and added to the ssh-agent. This key was created earlier during the Orka base image setup.